### PR TITLE
BubbleSorting method

### PR DESCRIPTION
**Pull request by Mohamed-js**
We have simulated the action of the native "sort" method and after this, we made the "bubble sort by" method.

### CODE REVIEW

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## :wave: :wave: Hello @Mohamed-js  and @RokoVarano. Thanks for submitting you project!

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] README is not professional
- [x] The names of the methods should be `bubble_sort`
- [x] Remove all `puts`/`prints` from the methods
- [x] The `test_results` method is not necessary for production please remove it.
- [x] Remove unnecessary files `.gitattributes` is not necessary for production or add them to `.gitignore`
- [x] `bubble_sort_by` method is missing
  - [x] should accept a block
  - [x] should raise an error if no block is given 

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      arr.delete_at(index + 2)
+      switched = true
+    end
+    index += 1
+    if switched
+      index = 0
+      switched = false
+    end
+  end
+  arr
```
Comment (arslanbisharat) [inline bubble_sort.rb]: - [ ] it should raise an error if no block is given

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Request changes 😳

I am very happy to see this project and I can see you worked hard on this project. There are only a few issues that we need to address and I am sure that you will end up with a perfect project

_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_

This is a great opportunity to learn. Microverse is here to help you. 

**Please have a look at the suggestions to get more understanding.**


- [ ] It was requested by previous TSE that bubble_sort_by should raise an error if no block is given

**Please make the changes and request a re-review for the given project**

If you get stuck somewhere, don't hesitate to reach me out or schedule s 1:1 meeting with a Full-time TSE. 

Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Request changes 😳

I am very happy to see this project and I can see you worked hard on this project. There are only a few issues that we need to address and I am sure that you will end up with a perfect project

_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_

This is a great opportunity to learn. Microverse is here to help you. 

**Please have a look at the suggestions to get more understanding.**


- [ ] You tried but the error is still present.
- [ ] Now you have Linter errors too. Please remove them

**Please make the changes and request a re-review for the given project**

If you get stuck somewhere, don't hesitate to reach me out or schedule s 1:1 meeting with a Full-time TSE. 

Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Request changes 😳

I am very happy to see this project and I can see you worked hard on this project. There are only a few issues that we need to address and I am sure that you will end up with a perfect project

_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_

This is a great opportunity to learn. Microverse is here to help you. 

**Please have a look at the suggestions to get more understanding.**


You are having the same error again and again please reach out to full-time TSE

**Please make the changes and request a re-review for the given project**


Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Review by descholar-ceo — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
## :clap: :clap: :clap: Congratulations @RokoVarano and @Mohamed-js  , your project is approved :tada: :tada: :tada: 
You have implemented all minimum requirements for this project to mark it as complete, it doesn't need more changes, you can go ahead and merge all of the changes on base branch,

Don't forget to fill the mandatory project completion form, and see you to next project,

Happy coding :computer: :computer: :computer: 

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Pull request merged**